cmd/broadcaster-cli/app/utxos/broadcast/broadcast.go

```diff
@@ -109,43 +110,66 @@ var Cmd = &cobra.Command{
 			return fmt.Errorf("failed to create client: %v", err)
 		}
 
-		fundingKeySets := make([]*keyset.KeySet, len(keyFiles))
-		for i, kf := range keyFiles {
+		rbs := make([]*broadcaster.RateBroadcaster, len(keySets))
+
+		wg := &sync.WaitGroup{}
 
-			fundingKeySet, _, err := helper.GetKeySetsKeyFile(kf)
+		var resultsPath string
+		if store {
+			network := "mainnet"
+			if isTestnet {
+				network = "testnet"
+			}
+			resultsPath = filepath.Join(".", fmt.Sprintf("results/%s-%s-rate-%d-batchsize-%d", network, time.Now().Format(time.DateTime), rateTxsPerSecond, batchSize))
+			err := os.MkdirAll(resultsPath, os.ModePerm)
 			if err != nil {
-				return fmt.Errorf("failed to get key sets: %v", err)
+				return err
 			}
-
-			fundingKeySets[i] = fundingKeySet
 		}
 
-		wocClient := woc_client.New(woc_client.WithAuth(wocApiKey), woc_client.WithLogger(logger))
+		_, cancel := context.WithCancel(context.Background())
+		defer cancel()
 
-		rateBroadcaster, err := broadcaster.NewMultiKeyRateBroadcaster(logger, client, fundingKeySets, wocClient, isTestnet,
-			broadcaster.WithFees(miningFeeSat),
-			broadcaster.WithCallback(callbackURL, callbackToken),
-			broadcaster.WithFullstatusUpdates(fullStatusUpdates),
-			broadcaster.WithBatchSize(batchSize),
-		)
-		if err != nil {
-			return fmt.Errorf("failed to create rate broadcaster: %v", err)
+		for i, keyset := range keySets {
```

Comment:
No need to iterate through the key sets. Just pass them to the multi key rate broadcaster


```

rateBroadcaster, err := broadcaster.NewMultiKeyRateBroadcaster(logger, client, keySets, wocClient, isTestnet,
			broadcaster.WithFees(miningFeeSat),
			broadcaster.WithCallback(callbackURL, callbackToken),
			broadcaster.WithFullstatusUpdates(fullStatusUpdates),
			broadcaster.WithBatchSize(batchSize),
		)
```

Reply:
This is done